README.md
# caja-mediainfo-tab
View media information from the properties tab

## Uses mediainfo lib (https://mediaarea.net/en/MediaInfo)

### It came to my attention that this extension was uploaded to Debian Sid. It's good to be noticed. It's good to be part of a big family. :smile:

## Installation

ppa:caldas-lopes/ppa

or

https://github.com/linux-man/caja-mediainfo-tab/releases

Use caja-mediainfo-tab-gtk2 if Caja < 1.18.

### Translations from MediaInfo(thanks to @seigneurfuo):

ar, be, bg, ca, cs, da, de, en, es, eu, fa, fr, gl, gr, hr, hu, hy, it, ja, ka, ko, lt, nl, pl, pt\_BR, pt, ro, ru, sk, sq, sv, th, tr, uk, zh\_CN, zh\_HK and zh\_TW



debian/changelog
caja-mediainfo-tab (1.0.4) focal; urgency=medium

  * Optional Char Wrap on long strings

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Thu, 02 Sep 2021 03:05:23 +0100

caja-mediainfo-tab (1.0.3) focal; urgency=medium

  * Mediainfo 3 dependency

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Fri, 25 Apr 2020 03:05:23 +0100

caja-mediainfo-tab (1.0.2) focal; urgency=medium

  * Python 3 dependencies

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Fri, 25 Apr 2020 02:08:23 +0100

caja-mediainfo-tab (1.0.1) bionic; urgency=medium

  * Utf-8 decode

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Fri, 24 May 2019 23:01:34 +0100

caja-mediainfo-tab (1.0) bionic; urgency=medium

  * Force plain text output
  * Call static options before opening file
  * Close MediaInfo object

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Sun, 05 May 2019 23:46:12 +0100

caja-mediainfo-tab (0.3) bionic; urgency=medium

  * Added an exclude list to hide specific tags

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Wed, 06 Mar 2019 12:00:25 +0100

caja-mediainfo-tab (0.2) bionic; urgency=medium

  * Translations from MediaInfo

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Mon, 10 Sep 2018 17:40:25 +0100

caja-mediainfo-tab (0.1) bionic; urgency=medium

  * Initial release

 -- Caldas Lopes <joao.caldas.lopes@gmail.com>  Mon, 04 Jun 2018 22:53:50 +0100


caja-mediainfo-tab-gtk2/caja-extension/caja-mediainfo-tab-gtk2.py
#!/usr/bin/python
#coding: utf-8

try:
    from urllib import unquote
except ImportError:
    from urllib.parse import unquote

from gi.repository import Caja, GObject, Gtk

from MediaInfoDLL import *

GUI = """
<interface>
  <requires lib="gtk+" version="2.0"/>
  <object class="GtkScrolledWindow" id="mainWindow">
    <property name="visible">True</property>
    <property name="can_focus">True</property>
    <property name="hscrollbar_policy">never</property>
    <child>
      <object class="GtkViewport" id="viewport1">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
      </object>
    </child>
  </object>
</interface>"""

class Mediainfo(GObject.GObject, Caja.PropertyPageProvider):

  def get_property_pages(self, files):

    if len(files) != 1:
      return

    file = files[0]
    if file.get_uri_scheme() != 'file':
      return

    if file.is_directory():
      return

    filename = unquote(file.get_uri()[7:])

    MI = MediaInfo()
    MI.Open(filename.decode("utf-8"))
    MI.Option_Static("Complete")
    info = MI.Inform().splitlines()
    if len(info) < 8:
      return

    self.property_label = Gtk.Label('Media Info')
    self.property_label.show()

    self.builder = Gtk.Builder()
    self.builder.add_from_string(GUI)

    self.mainWindow = self.builder.get_object("mainWindow")
    self.viewport = self.builder.get_object("viewport1")
    self.grid = Gtk.VBox()
    self.grid.show()
    self.viewport.add(self.grid)

    for line in info:
      box = Gtk.HBox(homogeneous=True, spacing=8)
      box.show()
      label = Gtk.Label()
      label.set_markup("<b>" + line[:41].strip() + "</b>")
      label.set_justify(Gtk.Justification.LEFT)
      label.set_alignment(0, 0.5)
      label.show()
      box.pack_start(label, True, True, 0)
      label = Gtk.Label()
      label.set_text(line[42:].strip())
      label.set_justify(Gtk.Justification.LEFT)
      label.set_alignment(0, 0.5)
      label.set_selectable(True)
      label.set_line_wrap(True)
      label.show()
      box.pack_start(label, True, True, 0)
      self.grid.pack_start(box, True, True, 0)

    return Caja.PropertyPage(name="CajaPython::mediainfo", label=self.property_label, page=self.mainWindow),



caja-extension/caja-mediainfo-tab.py
#!/usr/bin/python
#coding: utf-8

import locale, os

try:
  from urllib import unquote
except ImportError:
  from urllib.parse import unquote

from gi.repository import Caja, GObject, Gtk

from MediaInfoDLL3 import *

lang = locale.getdefaultlocale()[0]
locale_path = os.path.join(os.path.dirname(__file__), "caja-mediainfo-tab/locale")
locale_file = os.path.join(locale_path, lang+".csv")
if(not os.path.isfile(locale_file)):
  lang = lang.split("_")[0]
  locale_file = os.path.join(locale_path, lang+".csv")

excludeList = ["METADATA_BLOCK_PICTURE"]

GUI = """
<interface>
  <requires lib="gtk+" version="3.0"/>
  <object class="GtkScrolledWindow" id="mainWindow">
    <property name="visible">True</property>
    <property name="can_focus">True</property>
    <property name="hscrollbar_policy">never</property>
    <child>
      <object class="GtkViewport" id="viewport1">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <child>
          <object class="GtkGrid" id="grid">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="vexpand">True</property>
            <property name="margin_left">8</property>
            <property name="margin_right">8</property>
            <property name="margin_top">8</property>
            <property name="margin_bottom">8</property>
            <property name="row_spacing">4</property>
            <property name="column_spacing">16</property>
          </object>
        </child>
      </object>
    </child>
  </object>
</interface>"""

class Mediainfo(GObject.GObject, Caja.PropertyPageProvider):

  def get_property_pages(self, files):

    if len(files) != 1:
      return

    file = files[0]
    if file.get_uri_scheme() != 'file':
      return

    if file.is_directory():
      return

    filename = unquote(file.get_uri()[7:])

    try:
      filename = filename.decode("utf-8")
    except:
      pass

    MI = MediaInfo()
    MI.Option_Static("Complete")
    MI.Option_Static("Inform", "Nothing")
    MI.Option_Static("Language", "file://{}".format(locale_file))
    MI.Open(filename)
    info = MI.Inform().splitlines()
    MI.Close()
    if len(info) < 8:
      return

    self.property_label = Gtk.Label('Media Info')
    self.property_label.show()

    self.builder = Gtk.Builder()
    self.builder.add_from_string(GUI)

    self.mainWindow = self.builder.get_object("mainWindow")
    self.grid = self.builder.get_object("grid")

    top = 0
    for line in info:
      tag = line[:41].strip()
      if tag not in excludeList:
        label = Gtk.Label()
        label.set_markup("<b>" + tag + "</b>")
        label.set_justify(Gtk.Justification.LEFT)
        label.set_halign(Gtk.Align.START)
        label.show()
        self.grid.attach(label, 0, top, 1, 1)
        label = Gtk.Label()
        label.set_text(line[42:].strip())
        label.set_justify(Gtk.Justification.LEFT)
        label.set_halign(Gtk.Align.START)
        label.set_selectable(True)
        label.set_line_wrap(True)
        label.set_line_wrap_mode(2) #PANGO_WRAP_WORD_CHAR
        label.show()
        self.grid.attach(label, 1, top, 1, 1)
        top += 1

    return Caja.PropertyPage(name="CajaPython::mediainfo", label=self.property_label, page=self.mainWindow),

